src/components/v1/V1Project/ProjectActivity/eventElems/TapEventElem.tsx

```diff
@@ -57,6 +60,13 @@ export default function TapEventElem({
 
   if (!event) return null
 
+  const handleProject = (projectId: number, allocator: string) => {
+    if (allocator.toLowerCase() === V1_V3_ALLOCATOR_ADDRESS.toLowerCase()) {
+      return <V2V3ProjectHandleLink projectId={projectId} />
+    }
+    return <V1ProjectHandle projectId={projectId} />
+  }
```

Comment:
**suggestion** defining render functions like this within other components ("closures") can lead to rendering bugs. Could we move this outside of the `TapEventElem` component scope?

Reply:
Agreed! 

Should we, in this case, do inline conditioning to avoid render functions, or I can move render function outside of the component scope?

Reply:
i think inline sounds fine in this case! good call